C-Plus-Plus/cp/Two_Pointers_Method.cpp

```diff
@@ -0,0 +1,127 @@
+// Given a sorted array A, size of N ( N elements) and a target value X, find if there exists any pair of elements in the Array such that their total sum is equal to X.
+
+/*
+First we can solve this problem thorough a Naive Solution , that is a Brute force Solution which will be O(N * N)
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int ar[100009];
+
+bool Pair_exist_by_Brute_force(int ar[], int N, int X)
+{
+    for(int i = 0; i < N; i++)
+    {
+        for(int j = 0; j < N; j++)
+        {
+            if(i != j)
+            {
+                if(ar[i] + ar[j] == X)
+                    return true;
+                if(ar[i] + ar[j] > X)
+                    break;
+            }
+        }
+    }
+    return false;
+
+}
+
+
+int main()
+{
+	int N, X;
+	cout << "Input array size and target value\n";
+	cin >> N >> X;
+	for(int i = 0; i < N; i++)
+    {
+        cin >> ar[i];
+    }
+    bool solve = Pair_exist_by_Brute_force(ar , N , X);
+    if(solve)
+        cout << "Pair found and their total sum is equal to target value\n";
+    else
+        cout << "Pair doesn't exist in the array\n";
+}
+
+/*
+
+Standard Input and Output
+
+Input array size and target value
+6 70
+10 20 35 50 75 80
+Pair found and their total sum is equal to target value
+
+Time Complexity : O(N * N)
+Space Complexity : O(1)
+
+*/
+
+
+/*
+Secondly , we can optimize our solution by using Two pointers Method and thus time complexity will be reduced to O(N)
+Two pointers Method is a method where one pointer starts from beginning and other from the end and they proceed towards each other simultaneously
+
+*/
+#include <bits/stdc++.h>

```

Comment:
Why has a second code been added here? Do you understand that this would mess up badly?